frontend/src/components/VersionSelector/VersionSelector.tsx

```diff
@@ -0,0 +1,139 @@
+import { Button, Outline } from "@maykin-ui/admin-ui";
+import { FC, useId, useState } from "react";
+import { ZaakTypeVersion } from "~/types";
+
+import "./VersionSelector.css";
+
+interface VersionSelectorProps {
+  versions: ZaakTypeVersion[];
+  onVersionChange?: (versionId: string) => void;
+}
+
+/**
+ * A reusable component to select a version of a ZaakType.
+ * This component makes a few assumptions:
+ * - The versions are NOT sorted, we expect the backend to provide them in the correct order.
+ * - The current version is the one with a truthy `einde_geldigheid`.
+ * - The concept version is the one with `concept` set to true.
+ * - All other versions are considered historical versions.
+ *
+ * @param versions - An array of versions of a ZaakType as returned by the BFF.
+ * @param onVersionChange - An (optional) callback function to handle version changes.
+ * @returns A section with (collapsible) buttons to select a version.
+ * @example
+ * ```tsx
+ * <VersionSelector
+ *  versions={[
+ *  { uuid: "1", begin_geldigheid: "2023-01-01", einde_geldigheid: null, concept: false },
+ *  { uuid: "2", begin_geldigheid: "2024-01-01", einde_geldigheid: "2023-12-31", concept: false },
+ *  { uuid: "3", begin_geldigheid: "2024-02-01", einde_geldigheid: null, concept: true },
+ *  { uuid: "4", begin_geldigheid: "2024-03-01", einde_geldigheid: null, concept: false },
+ *  ]}
+ *  onVersionChange={(versionId) => console.log("Selected version:", versionId)}
+ *  />
+ * ```
+ *  */
+export const VersionSelector: FC<VersionSelectorProps> = ({
+  versions,
+  onVersionChange,
+}) => {
+  const id = useId();
+  const headingId = `version-selector-heading-${id}`;
+
+  const currentVersion = versions.find((v) => v.einde_geldigheid);
+  const conceptVersion = versions.find((v) => v.concept);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [selectedVersionId, setSelectedVersionId] = useState<
+    string | undefined
+  >(currentVersion?.uuid || currentVersion?.uuid);
+
+  const handleVersionChange = (versionId: string) => {
+    setSelectedVersionId(versionId);
+    onVersionChange?.(versionId);
+  };
+
+  const getVisibleVersions = () => {
+    // If expanded, show all versions
+    // If not expanded, show only the current and concept versions, and also the selected version
+    if (isExpanded) {
+      return versions;
+    }
+
+    return versions.filter((v) => {
+      return (
+        v.einde_geldigheid ||
```

Comment:
Is this correct? I think versions non-concept versions cannot overlap. So any previous (published) version should ahve an "einde_geldigheid" set (I think, please double check with @SilviaAmAm ).

If previous published versions have an einde_geldigheid, they should not always be shown. I think we should only show the current version (if available) and the future "conept" version (if available). So if not expanded:

- Show the last published version, only if the `begin_geldigheid` is < today and `einde_geldigheid` > today, if available.
- Show the concept last concept version, if availalble.
- Model the data restrictions in the Story.

Also can you work together with @SilviaAmAm to get the API value to use camelCase (and alter frontend accordingly).